lib/AST/Attr.cpp

```diff
@@ -1503,6 +1505,31 @@ void DifferentiableAttr::setOriginalDeclaration(Decl *decl) {
   OriginalDeclaration = decl;
 }
 
+bool DifferentiableAttr::hasComputedParameterIndices() const {
+  return ParameterIndicesAndBit.getInt();
+}
+
+IndexSubset *DifferentiableAttr::getParameterIndices() const {
```

Comment:
It seems more in line with (my understanding of) the request system philosophy to have callers make the requests directly instead of going through wrapper methods like this. Could you delete this method and make requests directly instead? Removing `Decl *decl` from the request makes this much easier.

If there are a huge number of callers of `getParameterIndices`, you could keep this for now to simplify the PR and then we can gradually migrate to making the request.

Reply:
Actually, my understanding is that wrappers around requests are desirable to keep caller code short; request evaluation is just an internal detail. I don't think there are downsides to using wrappers instead of direct results.

Examples:
* `ValueDecl::getInterfaceType()` [triggers a `InterfaceTypeRequest`](https://github.com/apple/swift/blob/master/lib/AST/Decl.cpp#L2892).
* `NominalTypeDecl::getStoredProperties()` [triggers a `StoredPropertiesRequest`](https://github.com/apple/swift/blob/2dddfcb190cdb88a43b54492e93ecc9f6049665e/lib/AST/Decl.cpp#L3631).
* More in `lib/AST/Decl.cpp`.

Let me know if that makes sense!